<launch>

    <!--- Sim Time -->
    <param name="/use_sim_time" value="true"/>

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lego_loam)/launch/test.rviz"/>

    <!-- Vehicle Model -->
    <include file="$(find vehicle_description)/launch/lexus.launch"/>

    <arg name="file_dir" default="$(find lego_loam)/map/"/>
    <!-- map to init transform -->
    <arg name="x" default="0.0"/>
    <arg name="y" default="0.0"/>
    <arg name="z" default="0.0"/>
    <arg name="roll" default="1.570795"/>
    <arg name="pitch" default="0.0"/>
    <arg name="yaw" default="1.570795"/>

    <!--- TF -->
    <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"
          args="0 0 0 1.570795   0        1.570795 /map    /camera_init 10"/>
    <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera"
          args="0 0 0 -1.570795 -1.570795 0        /camera /base_link   10"/>
    <node pkg="tf" type="static_transform_publisher" name="base_link_to_velodyne"
          args="0 0 0 -1.570795 -1.570795 0   /base_link /velo_left   10"/>
    <!--    <node pkg="tf" type="static_transform_publisher" name="base_link_to_imu"-->
    <!--          args="0 0 0 -1.570795 -1.570795 0   /base_link /imu   10"/>-->

    <!--- LeGO-LOAM -->
    <node pkg="lego_loam" type="imageProjection" name="imageProjection" output="screen"/>
    <node pkg="lego_loam" type="featureAssociation" name="featureAssociation" output="screen"/>
    <node pkg="lego_loam" type="mapOptmization" name="mapOptmization" output="screen"/>
    <node pkg="lego_loam" type="transformFusion" name="transformFusion" output="screen"/>

</launch>
